crypto/src/hash.rs

```diff
@@ -465,12 +465,8 @@ impl HashType {
             Err(FromBytesError::InvalidSize)
         } else {
             let mut hash = Vec::with_capacity(self.base58check_prefix().len() + data.len());
-            if matches!(self, Self::UnknownSignature) && data == [0; Self::Ed25519Signature.size()]
-            {
-                hash.extend(Self::Ed25519Signature.base58check_prefix());
-            } else {
-                hash.extend(self.base58check_prefix());
-            }
+
+            hash.extend(self.base58check_prefix());
```

Comment:
There's also a mirror check in `b58check_to_hash`, which seems entirely redundant: if you have an `edsig` you wouldn't generally try to decode it as `sig` in the first place.

I'm referring to this piece of code specifically:

```rust
        if let HashType::UnknownSignature = self {
            // zero signature is represented as Ed25519 signature
            if hash.len()
                == HashType::Ed25519Signature.size()
                    + HashType::Ed25519Signature.base58check_prefix().len()
            {
                let (prefix, hash) =
                    hash.split_at(HashType::Ed25519Signature.base58check_prefix().len());
                if prefix == HashType::Ed25519Signature.base58check_prefix()
                    && hash == [0; HashType::Ed25519Signature.size()]
                {
                    return Ok(hash.to_vec());
                }
            }
        } else ...
```

Reply:
done - removed this `if`